src/main/scala/io/iohk/ethereum/jsonrpc/TestService.scala

```diff
@@ -290,57 +319,86 @@ class TestService(
       .timeout(timeout.duration)
   }
 
+  /** Get the list of accounts of size _maxResults in the given _blockHashOrNumber after given _txIndex.
+    * In response AddressMap contains addressHash - > address starting from given _addressHash.
+    * nexKey field is the next addressHash (if any addresses left in the state).
+    * @see https://github.com/ethereum/retesteth/wiki/RPC-Methods#debug_accountrange
+    */
   def getAccountsInRange(request: AccountsInRangeRequest): ServiceResponse[AccountsInRangeResponse] = {
+    // This implementation works by keeping a list of know account from the genesis state
+    // It might not cover all the cases as an account created inside a transaction won't be there.
+
     val blockOpt = request.parameters.blockHashOrNumber
       .fold(number => blockchain.getBlockByNumber(number), blockHash => blockchain.getBlockByHash(blockHash))
 
     if (blockOpt.isEmpty) {
       AccountsInRangeResponse(Map(), ByteString(0)).rightNow
     } else {
-      val accountBatch = accountAddresses
-        .slice(accountRangeOffset, accountRangeOffset + request.parameters.maxResults.toInt + 1)
+      val accountBatch: Seq[(ByteString, Address)] = accountHashWithAdresses.view
+        .dropWhile { case (hash, _) => UInt256(hash) < UInt256(request.parameters.addressHash) }
+        .filter { case (_, address) => blockchain.getAccount(address, blockOpt.get.header.number).isDefined }
+        .take(request.parameters.maxResults + 1)
+        .to(Seq)
 
-      val addressesForExistingAccounts = accountBatch
-        .filter(key => blockchain.getAccount(Address(key), blockOpt.get.header.number).isDefined)
-        .map(key => (key, Address(crypto.kec256(Hex.decode(key)))))
+      val addressMap: Map[ByteString, ByteString] = accountBatch
+        .take(request.parameters.maxResults)
+        .map { case (hash, address) => hash -> address.bytes }
+        .to(Map)
 
       AccountsInRangeResponse(
-        addressMap = addressesForExistingAccounts
-          .take(request.parameters.maxResults.toInt)
-          .foldLeft(Map[ByteString, ByteString]())((el, addressPair) =>
-            el + (addressPair._2.bytes -> ByteStringUtils.string2hash(addressPair._1))
-          ),
+        addressMap = addressMap,
         nextKey =
           if (accountBatch.size > request.parameters.maxResults)
-            ByteStringUtils.string2hash(addressesForExistingAccounts.last._1)
+            accountBatch.last._1
           else UInt256(0).bytes
       ).rightNow
     }
   }
 
+  /** Get the list of storage values starting from _begin and up to _begin + _maxResults at given block.
+    * nexKey field is the next key hash if any key left in the state, or 0x00 otherwise.
+    *
+    * Normally, this RPC method is supposed to also be able to look up the state after after transaction
+    * _txIndex is executed. This is currently not supported in mantis.
```

Comment:
Do we have a ticket to track this lack of support? And are tests failing because of it?

Reply:
We have ETCM-784 and ETCM-758 which are related to this. I'll add a comment there. 

I did not find any test specifically failing because of this but I did not really look why other suites are failing. So I went with an iterative approach by implementing the minimum to make `vmArithmeticTest` pass.